### ISSUE TITLE
[ENH] Transformers that need conversion from nested pd.DataFrame to numpy arrays

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#197]
nested_univ is no longer going to be supported in aeon, switching fully to numpy. Some transformers still store things internally in this format. Nearly all of them just iterate and convert to numpy, so can just be converted to "X_inner_mtype": "numpy3D", and then adjusted when we facilitate unequal length. I'll list here as I find them (would search tags but doesnt work, see #182 )

- [x] DerivativeSlopeTransformer _extract.py  #392 
- [x] PlateauFinder in _extract.py #392 
- [x] RandomIntervalFeatureExtractor in compose.py #391
- [x] RandomIntervalSegmenter  in segment.py #391 
- [x] IntervalSegmenter  in segment.py #391
- [x] TSInterpolator (why is it named this? We do not use TS suffix elsewhere?) interpolate.py #388 
- [x] PaddingTransformer (again, may refactor the name) in padder.py  #364
- [x] TruncationTransformer in truncation.py #364 
transformations/Panel/reduce.py
- [x] SFA
- [x] SlidingWindowTransformer
- [x] DerivativeSlopeTransformer #261  
- [x] Catch22
- [x] Catch22Wrapper
- [x] DWTTransformer
- [x] HOG1DTransformer #261 
- [x] SlopeTransformer
- [x] TruncationTransformer
- [x] _TSFreshFeatureExtractor, TSFreshFeatureExtractor, TSFreshRelevantFeatureExtractor
- [x] PAA #269 
- [x] All the rocket transformers #209 
- [x] ShapeTranformer #250 

Leaving these
Tabularizer reduce.py
its embedded in the reduction stuff in forecasting, so I dont want to mess with it
ColumnTransformer
can sort it out when we address #171

basically need to switch from 
        "X_inner_mtype": "nested_univ",  # which mtypes do _fit/_predict support for X?
to this
        "X_inner_mtype": "numpy3D",  
and obviously make it all work and update any tests. Also a good time to add examples. Also need to do this in the constructor to stop nonsense
        super(PAA, self).__init__(_output_convert=False)

then treat X as numpy in fit and transform. Some much easier than others, any that work with unequal length (e.g. truncate and pad) leave for now because need to work with lists of numpy

**Comment by aiwalter** [comment]
Actually it seems nested pandas is still working with pandas 2.0.0

**Comment by TonyBagnall** [comment]
still want to get rid of them :)

**Comment by lmmentel** [comment]
Thanks for the summary @TonyBagnall! I think it would be a good move to drop the nested, but could we discuss actual consequences and alternatives?

**Comment by TonyBagnall** [comment]
sure @lmmentel happy to discuss, shall we have a developers meeting on the issue?

**Comment by TonyBagnall** [comment]
but to clarify, this is all internal, it will not effect anything external. The base class will just do the conversions. We are not dropping nested_univ support, just storing things internally a numpy arrays to avoid unnecessary conversions with the standard use cases

**Comment by TonyBagnall** [comment]
Its done, enjoying closing this one :) I'll raise a separate issue for 
Tabularizer reduce.py and update #171 for ColumnTransformer
